test-cypress/integration/explore/graphs.overview.spec.js

```diff
@@ -1,114 +1,131 @@
+    function verifyGraph(graphName, exists = true) {
+        cy.get(EXPORT_GRAPHS_TABLE_ID).should(`${!exists ? 'not.' : ''}contain`, graphName);
+    }
 
+    function selectPage(number) {
+        return cy.get('div[paginations] ul li a').eq(number).click();
+    }
 
-    it('Test graphs overview pagination - test items per page', function () {
+    function selectItemFromMenu(number) {
         cy.get('[class="btn btn-secondary btn-sm dropdown-toggle"]').click();
-        cy.get('[class="dropdown-menu small"] li').eq(1).click();
-        cy.verifyGraphExists("urn:19");
-        cy.get('[class="btn btn-secondary btn-sm dropdown-toggle"]').click();
-        cy.get('[class="dropdown-menu small"] li').eq(2).click();
-        cy.verifyGraphExists("urn:1");
-        cy.verifyGraphExists("urn:21");
+        cy.get('[class="dropdown-menu small"] li').eq(number).click();
+    }
+
+    function confirmDelete() {
+        cy.get('.modal-footer .confirm-btn').click();
+        cy.get('.modal').should('not.be.visible');
+    }
+
+    context('Test graphs overview pagination', () => {
+        it('Should be visible', () => {
+            cy.get('div[paginations]')
+                .should('be.visible')
+                .and('contain', '3');
+            verifyGraph('The default graph');
+        });
+
+        it('Should switch pages', () => {
+            // Switch through pages and verify that the respective pager button is active.
+            selectPage(2).should('contain', '2')
+                .closest('li').should('be', 'active');
+            verifyGraph('urn:11');
+
+            selectPage(3).should('contain', '3')
+                .closest('li').should('be', 'active');
+            verifyGraph('urn:21');
+        });
     });
 
-    it('Test graphs overview search', function () {
-        cy.wait(1000);
-        cy.get('.form-control').type(graphToSearch);
-        cy.verifyGraphExists(graphToSearch);
-        cy.get("tbody").should('not.contain', "urn:11");
+    it('Test items per page switch menu', () => {
+        // Change number of items per page from the menu and verify that proper rows are present in
+        // the table.
+        selectItemFromMenu(1);
+        verifyGraph('urn:19');
 
+        selectItemFromMenu(2);
+        verifyGraph('urn:1');
+        verifyGraph('urn:21');
     });
 
-    it('Delete graph', function () {
-        cy.wait(1000);
-        cy.get('tbody tr a')
-            .each(($el, index) => {
-                if ($el.attr('title') === graphToDelete) {
-                    cy.get('[class="fa fa-trash-o fa-lg"]').eq(index).click({force: true});
-                    cy.get('.modal-content button.btn.btn-primary').click({force: true});
-                }
-            });
-        cy.get("tbody").should('not.contain', graphToDelete);
+    it('Test graphs overview search', () => {
+        // Default items/rows per page is 10.
+        verifyVisibleGraphsCountIs(10);
```

Comment:
`verifyVisibleGraphsCount` ?

Reply:
Somehow this seemed more natural to read to me :)